sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -413,6 +413,25 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
     )
   }
 
+  test("arrays_overlap function") {
+    val df = Seq(
+      (Seq[Option[Int]](Some(1), Some(2)), Seq[Option[Int]](Some(-1), Some(10))),
+      (Seq.empty[Option[Int]], Seq[Option[Int]](Some(-1), None)),
+      (Seq[Option[Int]](Some(3), Some(2)), Seq[Option[Int]](Some(1), Some(2)))
+    ).toDF("a", "b")
+
+    val answer = Seq(Row(false), Row(null), Row(true))
+
+    checkAnswer(df.select(arrays_overlap(df("a"), df("b"))), answer)
+    checkAnswer(df.selectExpr("arrays_overlap(a, b)"), answer)
+
+    checkAnswer(sql("select arrays_overlap(array(1, 2, 3), array('a', 'b', 'c'))"), Row(false))
+
```

Comment:
Do we add a test like this?
```
val df = Seq((null, null)).toDF("a", "b")
val ans = ...
checkAnswer(df.select(array_overlap($"a", $"b")), ans)
checkAnswer(df.selectExpr("array_overlap(a, b)"), ans)
```

Reply:
this case is already covered here: https://github.com/apache/spark/pull/21028/files#diff-d31eca9f1c4c33104dc2cb8950486910R136, am I right?

Reply:
IIUC, I think no. My comment is talking about `null` handling in Dataframe API.
[Other operations](https://github.com/apache/spark/pull/21028/files#diff-8e1a34391fdefa4a3a0349d7d454d86fR396) aslo perform tests with `null` in DataFrame API and primitive.

Reply:
Oh, now I see what you mean. I can add it, but it seems useless to me. This function accepts only `Array`s so any other type (`NullType` included) throws an `AnalysisException`. In `array_contains` is different since we the second argument can be anything and so makes sense to check the behavior of `NullType` which is handled differently from the others. Do you agree?

Reply:
I think that to add this makes sense to explicitly ensure `so any other type (NullType included) throws an AnalysisException`.

Reply:
Honestly I don't see its utility but I see also no harm in introducing it, so if you think it is a added value, I think it is fine to add it. So I just added it, thanks.

Reply:
Thank you.